src/solvers/lowering/byte_operators.cpp

```diff
@@ -1819,7 +1819,8 @@ static exprt lower_byte_update_struct(
     {
       offset = from_integer(0, src.offset().type());
       INVARIANT(
-        value_as_byte_array.operands().size() > -*offset_bytes,
```

Comment:
Would the byte array have a type reflecting its constant size? If so how about using the type-size which would work for comprehensions too?

Reply:
If it's an array comprehension then it would almost certainly be of non-constant size, and therefore couldn't be evaluated at this time.